### PR TITLE
Added option to log samples (but not log_pdfs) to disk during MCMCSampling

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #542 

Current solution is easy and is compatible with `io.load_samples()`. Would be good to have `log_pdf` in the files too though, but would probably require changes to MCMCSampling ask-and-tell interface.

@ben18785 what do you think? Is it a common request to have the `log_pdf` values of the samples stored as well? Sounds sensible although I haven't seen any examples.

### CODE REVIEW

File: pints/_mcmc/__init__.py [hunk 1]
```diff
@@ -651,6 +728,44 @@ def set_max_iterations(self, iterations=10000):
                     'Maximum number of iterations cannot be negative.')
         self._max_iterations = iterations
 
+    def set_output_files(self, chain_file, evaluation_file):
```
Comment (martinjrobins) [inline pints/_mcmc/__init__.py]: Might be nicer to split this into two functions, `set_output_chain_file` and `set_output_evaluation_file`.

Reason being is that the name `set_output_files` is not very descriptive, does it refer to the logging perhaps?, and if you come across this in your code

```python
sampler.set_output_files(filename, None)
```

you have no idea what you are outputing....

Comment (MichaelClerx) [reply pints/_mcmc/__init__.py]: Fair enough. I want to avoid confusion with the other _log_ methods though (which have nothing to do with this!)

Comment (martinjrobins) [reply pints/_mcmc/__init__.py]: i was thinking that having the words `chain` and `evaluation` might further distinguish it from the logging functions...

Comment (MichaelClerx) [reply pints/_mcmc/__init__.py]: Ok, have updated them now!

Comment (mirams) [reply pints/_mcmc/__init__.py]: Not sure that `evaluation` is very clear, I'd be expecting something more like `target` or `target_value`?

Comment (MichaelClerx) [reply pints/_mcmc/__init__.py]: Well we don't call it target anywhere else, plus the target is surely the distribution itself, not a single point within it? Happy to change it to say `set_logpdf_filename(...)`

**Comment by MichaelClerx** [comment]
@mirams I've changed it to `set_log_pdf_filename` is that better for you?

**Comment by mirams** [comment]
Yes! Thanks!

**Pull request merged**